analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicContext.cs

```diff
@@ -51,10 +51,4 @@ public SymbolicContext(IOperationWrapperSonar operation, ProgramState state, boo
 
     public SymbolicContext WithState(ProgramState newState) =>
         State == newState ? this : new(Operation, newState, IsLoopCondition, VisitCount, CapturedVariables);
-
-    public bool HasConstraint(IOperation operation, SymbolicConstraint constraint) =>
-        State[operation]?.HasConstraint(constraint) is true;
-
-    public bool HasConstraint(ISymbol symbol, SymbolicConstraint constraint) =>
-        State[symbol]?.HasConstraint(constraint) is true;
```

Comment:
I find these shortcuts pretty handy in the codebase. I want to keep them.

Reply:
One of my ideas turned out to be bad, so this refactor doesn't make much sense. My new approach is to move these shortcuts to `ProgramState` right now and make them null-safe (for `operation`/`symbol`). We can keep them in `SymbolicContext` if you want, but that seems redundant.

Reply:
ProgramState is fine